src/main/java/org/generationcp/middleware/service/impl/study/StudyEntryServiceImpl.java

```diff
@@ -47,14 +58,23 @@ public class StudyEntryServiceImpl implements StudyEntryService {
 	@Resource
 	private OntologyDataManager ontologyDataManager;
 
+	@Resource
+	private GermplasmListService germplasmListService;
+
+	@Resource
+	private PedigreeService pedigreeService;
+
+	@Resource
+	private CrossExpansionProperties crossExpansionProperties;
+
 	private final DaoFactory daoFactory;
 
+	// TODO: remove ENTRY_NO
 	private static final List<Integer> FIXED_GERMPLASM_DESCRIPTOR_IDS = Lists
-		.newArrayList(TermId.ENTRY_CODE.getId(), TermId.DESIG.getId(), TermId.ENTRY_NO.getId(), TermId.GID.getId());
+		.newArrayList(TermId.DESIG.getId(), TermId.ENTRY_NO.getId(), TermId.GID.getId(), TermId.IMMEDIATE_SOURCE_NAME.getId());
```

Comment:
@lik-83 @cuenyad 
1. Is the TODO still valid? I can see ENTRY_NO in the list
2. Why is IMMEDIATE_SOURCE_NAME in this list?

Reply:
I create this https://ibplatform.atlassian.net/browse/IBP-5793 for a cleanup.